### PR TITLE
Emiliano/210

### PR DESCRIPTION
**Pull request by emilian0**
Implementing Morph Targets as discussed in #210. 
@lexaknyazev all yours!

### CODE REVIEW

File: specification/2.0/README.md [hunk 1]
```diff
@@ -594,6 +594,41 @@ Valid attribute semantic property names include `POSITION`, `NORMAL`, `TEXCOORD`
 
 > **Implementation note:** Each primitive corresponds to one WebGL draw call (engines are, of course, free to batch draw calls). When a primitive's `indices` property is defined, it references the accessor to use for index data, and GL's `drawElements` function should be used. When the `indices` property is not defined, GL's `drawArrays` function should be used with a count equal to the count property of any of the accessors referenced by the `attributes` property (they are all equal for a given primitive).
 
+#### Morph Targets
+
+Morph Targets are defined by extending the Mesh concept.
+A Morph Target is a morphable Mesh where primitives' attributes are obtained by adding the original attributes to a weighted sum of targets attributes (this operation corresponds to COLLADA's `RELATIVE` blending method).
```
Comment (lexaknyazev) [inline specification/2.0/README.md]: It would be better to show equation than use external COLLADA reference.

File: specification/2.0/README.md [hunk 2]
```diff
@@ -594,6 +594,41 @@ Valid attribute semantic property names include `POSITION`, `NORMAL`, `TEXCOORD`
...
+Morph Targets are implemented via the `targets` property defined in the Mesh `primitives`. Each target in the `targets` array is a dictionary mapping a primitive attribute to a Morph Target displacement, currently only two attributes ('POSITION' and 'NORMAL') are supported. All primitives are required to list the same number of `targets` in the same order.
```
Comment (lexaknyazev) [inline specification/2.0/README.md]: > Each target ... mapping a primitive attribute to a Morph Target displacement

... mapping a primitive attribute to an accessor containing Morph Target displacement data

File: specification/2.0/README.md [hunk 3]
```diff
@@ -594,6 +594,41 @@ Valid attribute semantic property names include `POSITION`, `NORMAL`, `TEXCOORD`
...
+                    "POSITION": 33,
+                },
+                {
+                    "NORMAL": 45,
+                    "POSITION": 43,
+                },
+            ]
+        }
+    ],
+    "weights": [0, 0.5]
```
Comment (lexaknyazev) [inline specification/2.0/README.md]: It's still not quite obvious to me, why do we have two ways of defining initial morph state: `mesh.weights` and `node.weights`. Just asking.

File: specification/2.0/schema/mesh.primitive.schema.json [hunk 4]
```diff
@@ -31,6 +31,14 @@
             "default" : 4,
             "gltf_detailedDescription" : "The type of primitives to render. All valid values correspond to WebGL enums."
         },
+        "targets" : {
+            "type" : "array",
+            "description" : "An array of Morph Targets, each  Morph Target is a dictionary mapping attributes (only "POSITION" and "NORMAL" supported) to their deviations in the Morph Target.",
```
Comment (lexaknyazev) [inline specification/2.0/schema/mesh.primitive.schema.json]: Double-quotes must be escaped here.

Comment (emilian0) [inline specification/2.0/schema/mesh.primitive.schema.json]: fixed, thanks

File: specification/2.0/README.md (hunk 3 above)
Comment (emilian0) [inline specification/2.0/README.md]: `mesh.weights` are the Morph Targets default weights. 
`node.weights` are the instantiated Morph Targets weights (that override the weights of the mesh). 
Say that you have a Morph Target, it is a character face. You want to instantiate it multiple times in your scene, and you want to instantiate it as a smiling face. Then just set the default `mesh.weights` accordingly (make it smile!) and don't set the instantiation weights `node.weights` when you instantiate it within a node. 
Makes sense?

File: specification/2.0/README.md (hunk 2 above)
Comment (emilian0) [inline specification/2.0/README.md]: fixed, thanks

File: specification/2.0/README.md (hunk 1 above)
Comment (emilian0) [inline specification/2.0/README.md]: addressed. Thanks

**Comment by emilian0** [comment]
@lexaknyazev I addressed your concerns, back to you. Thanks

**Pull request merged**